Source/JavaScriptCore/jsc.cpp

```diff
@@ -3724,7 +3730,9 @@ static void runWithOptions(GlobalObject* globalObject, CommandLine& options, boo
             if (isModule) {
                 // If necessary, prepend "./" so the module loader doesn't think this is a bare-name specifier.
                 fileName = isAbsolutePath(fileName) || isDottedRelativePath(fileName) ? fileName : makeString('.', pathSeparator(), fileName);
-                promise = loadAndEvaluateModule(globalObject, fileName, jsUndefined(), jsUndefined());
+                Identifier resolvedIdentifier = globalObject->moduleLoader()->resolve(globalObject, jsString(vm, fileName), jsUndefined(), jsUndefined());
+                RETURN_IF_EXCEPTION(scope, void());
+                promise = globalObject->moduleLoader()->fetchModuleAndEvaluate(globalObject, resolvedIdentifier, jsUndefined(), jsUndefined());
```

Comment:
why we need `fetchModuleAndEvaluate` here since the source is from file?

Reply:
Before this change, `loadAndEvaluateModule` (of `Completion.cpp`) handled both 1) file name and 2) file source cases, which was confusing.

This patch makes a clear distinction: `fetch*` methods for file name cases like this one, while resorting to `load*` methods when file source is present.

I will make sure to note this API improvement in the ChangeLog.